dvc/stage/__init__.py

```diff
@@ -252,13 +252,36 @@ def is_checkpoint(self):
         """
         return any(out.checkpoint for out in self.outs)
 
-    @property
-    def env(self) -> Env:
+    def _read_env(self, out, checkpoint_func=None) -> Env:
+        env: Env = {}
+        if out.live:
+            from dvc.env import DVCLIVE_PATH, DVCLIVE_REPORT, DVCLIVE_SUMMARY
+            from dvc.output import BaseOutput
+            from dvc.schema import LIVE_PROPS
+
+            env[DVCLIVE_PATH] = str(out.path_info)
+            if isinstance(out.live, dict):
+                config = project(out.live, LIVE_PROPS)
+
+                env[DVCLIVE_SUMMARY] = str(
+                    int(config.get(BaseOutput.PARAM_LIVE_SUMMARY, True))
+                )
+                env[DVCLIVE_REPORT] = str(
+                    int(config.get(BaseOutput.PARAM_LIVE_REPORT, True))
+                )
+        elif out.checkpoint and checkpoint_func:
+            from dvc.env import DVC_CHECKPOINT, DVC_ROOT
+
+            env.update({DVC_CHECKPOINT: "1", DVC_ROOT: self.repo.root_dir})
```

Comment:
This isn't because of your changes, but looking at this now, I think we should be setting `DVC_ROOT` all the time in the calling `env()` function. (`DVC_CHECKPOINT` should still only be set inside this `if` case)

For experiments that are run in temp-dirs (and not only checkpoint experiments), it will always be useful for user's code to be able to see where it is being run, since they may need to work with relpaths from that directory (instead of relpaths inside their git/dvc repo workspace). And outside of the experiment context, it won't hurt us to set `DVC_ROOT` for regular `repro` runs as well.

Reply:
Good point!